client/src/components/TextCheckboxGroupField/TextCheckboxGroupField.scss

```diff
@@ -19,3 +19,8 @@
     }
   }
 }
+
+// hide .show-title in the Archives ModelAdmin only
+.cms-edit-form[action*='admin/archive/'] .show-title {
+  display: none;
```

Comment:
...? Can you not just change the readonly transformation to remove this? This feels really hacky.